docs: add note about keeping same jina versions (#5239)
jina-ai/jina: docs/how-to/docker-compose.md

> Jina will generate a `docker-compose.yml` configuration file that you can use directly with 
> `docker-compose` and corresponds to your `Flow`, avoiding the overhead of manually defining all the services needed for the `Flow`.

Reviewer: ```suggestion
Jina will generate a `docker-compose.yml` configuration file corresponding with your Flow. You can use this directly with 
Docker Compose, avoiding the overhead of manually defining all the services needed for the Flow.
```

````{admonition} Use docker based Executors
:class: caution
All Executors in the Flow should be used with `jinahub+docker://...` or `docker://...`.
````

````{admonition} Health check available from 3.1.3
:class: caution

> If you are using Executor which rely on docker image built with a jina version prior to 3.1.3, please remove the 
> health check from the dump yaml file as they are only compatible with 3.1.3+ otherwise your docker compose services will 
> always be `unhealthy`

Reviewer: ```suggestion
If you use Executors that rely on Docker images built with a version of Jina prior to 3.1.3, remove the 
health check from the dumped YAML file, otherwise your Docker Compose services will 
always be "unhealthy."
```

````

````{admonition} Matching jina versions
:class: caution

> If you alter the docker compose generated file to adapt the docker images used in the solution, try to make sure that all the services included

Reviewer: ```suggestion
If you change the Docker images in your Docker Compose generated file, ensure that all the services included in
```

> the gateway are built with the same jina version to guarantee compatibility.

Reviewer: ```suggestion
the gateway are built with the same Jina version to guarantee compatibility.
```

Capitalization :angry: 

Also, do we capitalize "gateway"? I'm unclear on this

Author: I have no opinion on this, I think we should

Reviewer: Fine by me :)

````
## Example: Indexing and searching images using CLIPEncoder and ANNLiteIndexer

> To follow this how-to, you should first ensure that [`Docker Compose`](https://docs.docker.com/compose/install/) is installed locally.

Reviewer: ```suggestion
Install [`Docker Compose`](https://docs.docker.com/compose/install/) locally to follow this how-to.
```
